### ISSUE TITLE
No test set parameter inside the setup function

### ISSUE THREAD

**Issue by vishal-subedi** [pycaret/pycaret#73]
The setup function obviously has the training and testing split size but lets say if a candidate wants to provide a separate training and testing dataset instead of passing a complete dataset and splitting it posterior to that; there is no parameter for this purpose.

**Comment by pycaret** [comment]
@vishal-subedi  Thanks.

So I am thinking why would somebody have a testset separately. For a Kaggle kind of scenario your testset doesn't have label so they are used under predict_model to generate predictions. Other scenario I could think of is where somebody wants to shuffle the data outside of pycaret.

What are your thoughts and use-case behind this idea. I like the idea very much.

**Comment by vishal-subedi** [comment]
Lets say I am working on a credit based dataset. 
Now, I currently have only 2 datasets as of today viz the train and test. Now, the testing data is very small so I want the training as well as tuning to be done on the train data itself. But the problem here is that the parameter train_size will split my training set itself into further training and hold-out set, which further reduces the accuracy. Now, lets say I join my training and testing data into a single dataset without any shuffle, and I know what percent of the whole data consists of test data, so I pass that percent into the train_size . This approach works to a certain extent. What I mean by that is the create_model creates and trains it on the training set and the plots show the evaluation on the retained hold-out set. But once the model is finalized using finalize_model(), the entire dataset including the test/hold-out set is used for training; which I was trying to avoid. 
Had there been a single parameter call test_data as well, all these problems could be solved easily

**Comment by moezali1** [comment]
Seems like a good idea. Would you like to work on this? This would require change in setup function for classification and regression. 

You can send the PR on dev-1.0.1 branch.

**Comment by vishal-subedi** [comment]
Ya sure, will try my best to come up with a solution, although I am new to this platform. Will take some time to analyze the basic functions.

**Comment by HiroTamori** [comment]
Hello, I found this issue on another case.

I'm trying to analyze time-series data separated in 10-minute time windows sliding 1-minute for calculating values included in each window like this picture below;
![time-slide-window](https://user-images.githubusercontent.com/10972221/82035384-69c86880-96da-11ea-8d22-03829854bcc1.png)

As the picture shows, time window 0 -  9 are overlapping and  1 -  10,  2 -  11,... also.
To avoid the contamination between train data set and test data set, I want to separate the first N windows for training and remaining ones for test.

But PyCaret's setup() function has only arguments for assign whole data set `(data)` and ratio of the number of training data `(train_size)`. 

Time series analysis often has the case like this.

**Comment by pycaret** [comment]
@Yard1 With the new structure it is going to be a very simple change. We can add a new parameter in the `setup` function for `regression` and `classification`. Let's say the parameter is called `test_data` which takes a pandas data frame. The default should be `None`  but when `test_data` is provided and it matches the `data` and all other exception checkings, there is no need for `train_test_split` as we already have `train` and `test` separated out. The only thing then needed is to split X and y which is in our control.  I think this will just add more flexibility for users. What are your thoughts?

**Comment by Yard1** [comment]
Sounds good.

**Comment by pycaret** [comment]
@vishal-subedi @HiroTamori 

This issue has been fixed in the `pycaret-nightly` release. Please install pycaret-nightly using `pip install pycaret-nightly`. Feel free to open a new issue, in case the problem persists. This fix will be published to PyPi as `pycaret==2.2.0` on October 28, 2020.
